gobblin-data-management/src/main/java/org/apache/gobblin/data/management/conversion/hive/task/HiveMaterializerQueryGenerator.java

```diff
@@ -0,0 +1,228 @@
+    this.partitionsDMLInfo = Maps.newHashMap();
+    HiveConverterUtils.populatePartitionInfo(conversionEntity, partitionsDDLInfo, partitionsDMLInfo);
+    this.destinationTableMeta = HiveConverterUtils.getDestinationTableMeta(outputDatabaseName,
+        outputTableName, workUnitState).getLeft();
+  }
+
+  @Override
+  public List<String> generateQueries() {
+
+    List<String> hiveQueries = Lists.newArrayList();
+
+    Preconditions.checkNotNull(this.workUnit, "Workunit must not be null");
+    EventWorkunitUtils.setBeginDDLBuildTimeMetadata(this.workUnit, System.currentTimeMillis());
+
+    HiveConverterUtils.createStagingDirectory(fs, conversionConfig.getDestinationDataPath(),
+        conversionEntity, this.workUnitState);
+
+    // Create DDL statement for table
+    String createStagingTableDDL =
+        HiveConverterUtils.generateCreateDuplicateTableDDL(
+            inputDbName,
+            inputTableName,
+            outputStagingTableName,
+            outputStagingDataLocation,
+            Optional.of(outputDatabaseName));
+    hiveQueries.add(createStagingTableDDL);
+    log.debug("Create staging table DDL:\n" + createStagingTableDDL);
+
+    // Create DDL statement for partition
+    if (partitionsDMLInfo.size() > 0) {
+      List<String> createStagingPartitionDDL =
+          HiveAvroORCQueryGenerator.generateCreatePartitionDDL(outputDatabaseName,
+              outputStagingTableName,
+              stagingDataPartitionLocation,
+              partitionsDMLInfo);
+
+      hiveQueries.addAll(createStagingPartitionDDL);
+      log.debug("Create staging partition DDL: " + createStagingPartitionDDL);
+    }
+
+    hiveQueries.add("SET hive.exec.dynamic.partition.mode=nonstrict");
+
+    String insertInStagingTableDML =
+        HiveConverterUtils
+            .generateTableCopy(
+                inputTableName,
+                outputStagingTableName,
+                Optional.of(conversionEntity.getHiveTable().getDbName()),
+                Optional.of(outputDatabaseName),
+                Optional.of(partitionsDMLInfo),
+                Optional.absent(),
+                Optional.absent());
+    hiveQueries.add(insertInStagingTableDML);
+    log.debug("Conversion staging DML: " + insertInStagingTableDML);
+
+    log.info("Conversion Query {}",  hiveQueries);
+
+    EventWorkunitUtils.setEndDDLBuildTimeMetadata(workUnit, System.currentTimeMillis());
+    return hiveQueries;
+  }
+
+  public QueryBasedHivePublishEntity generatePublishQueries() throws DataConversionException {
+
+    QueryBasedHivePublishEntity publishEntity = new QueryBasedHivePublishEntity();
+    List<String> publishQueries = publishEntity.getPublishQueries();
+    Map<String, String> publishDirectories = publishEntity.getPublishDirectories();
+    List<String> cleanupQueries = publishEntity.getCleanupQueries();
+    List<String> cleanupDirectories = publishEntity.getCleanupDirectories();
+
+    String createFinalTableDDL =
```

Comment:
This should be optional. Sometimes we don't actually want to create a new table, just the data.

Reply:
The create table command is like 'CREATE TABLE IF NOT EXISTS...'
In the job file output table name can be kept as source table name (by default), and then it will not create a new table. In test flows, target table can be set to '${source}_test'